daemon/openrazer_daemon/hardware/mouse.py

```diff
@@ -15,6 +15,50 @@
     set_left_brightness as _set_left_brightness, set_right_brightness as _set_right_brightness
 
 
+class RazerViperMini(__RazerDeviceSpecialBrightnessSuspend):
+    """
+    Class for the Razer Viper Mini
+    """
+    EVENT_FILE_REGEX = re.compile(r'.*Razer_Razer_Viper_Mini-if0(1|2)-event-kbd')
+
+    USB_VID = 0x1532
+    USB_PID = 0x008A
+    METHODS = ['get_device_type_mouse', 'max_dpi', 'get_dpi_xy', 'set_dpi_xy', 'get_poll_rate', 'set_poll_rate', 'get_logo_brightness', 'set_logo_brightness',
+               # Underglow/Logo use LOGO_LED
+               'set_logo_static_naga_hex_v2', 'set_logo_spectrum_naga_hex_v2', 'set_logo_none_naga_hex_v2', 'set_logo_reactive_naga_hex_v2', 'set_logo_breath_random_naga_hex_v2', 'set_logo_breath_single_naga_hex_v2', 'set_logo_breath_dual_naga_hex_v2',
+               # Can set Underglow/LOGO with custom
+               'set_custom_effect', 'set_key_row']
```

Comment:
If the device has custom effect support, it needs HAS_MATRIX = True and the matrix dimensions set

Reply:
I've removed the references to custom effect support here as well as in the fake driver